src/components/user/HomeMap.js

```diff
@@ -1,10 +1,31 @@
-import React from 'react';
-
-import { MapContainer, TileLayer, Marker, Popup } from '@monsonjeremy/react-leaflet';
+import React, { useEffect, useState } from 'react';
+import { MapContainer, TileLayer, Marker, Popup, useMap } from '@monsonjeremy/react-leaflet';
 import 'leaflet/dist/leaflet.css';
 
-const HomeMap = () => {
-  const position = [-34.907, -56.203];
+const DEFAULT_LONGITUDE = -34.907;
+const DEFAULT_LATITUDE = -56.203;
+
+const position = [DEFAULT_LATITUDE, DEFAULT_LONGITUDE];
+
+function HomeMap() {
+  function LocationMarker() {
+    const [position, setPosition] = useState(null);
+
+    const map = useMap();
+
+    useEffect(() => {
+      map.locate().on('locationfound', function(e) {
```

Comment:
You might want to consider the `locationerror` event as well and respond to it, right?

Reply:
As a suggest (what I used) you can use the hook provided by `react-leaflet` : `useMapEvents`. I think that is a better aproach but only as a personal opinion, u can check the documentation for that here : 
https://react-leaflet.js.org/docs/api-map/#usemapevents

Reply:
Alright! I'll take a look into it so I can take advantage of those